src/org/labkey/test/params/experiment/SampleTypeDefinition.java

```diff
@@ -211,6 +223,10 @@ protected String getKind()
         {
            options.put("autoLinkTargetContainerId", getAutoLinkDataToStudy());
         }
+        if(getLinkedDatasetCategory() != null){
```

Comment:
Tiny nitpick: Looks like both `if(getAutoLinkDataToStudy() != null)` and `if(getLinkedDatasetCategory() != null)` need a extra space, so they look like `if (getAutoLinkDataToStudy() != null)` and `if (getLinkedDatasetCategory() != null)`